### ISSUE TITLE
Extended.deploymentconfigs when run iteratively [Conformance] should only deploy the last deployment

### ISSUE THREAD

**Issue by csrwng** [openshift/origin#13575]
Conformance test flake:

https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_gce/650/

```
[It] should only deploy the last deployment
  /openshifttmp/openshift/build-rpm-release/tito/rpmbuild-originwPIlFK/BUILD/origin-3.6.0/_output/local/go/src/github.com/openshift/origin/test/extended/deployments/deployments.go:137
Mar 29 11:15:25.741: INFO: Running 'oc create --config=/tmp/extended-test-cli-deployment-xwxwb-4hs6w-user.kubeconfig --namespace=extended-test-cli-deployment-xwxwb-4hs6w -f /tmp/fixture-testdata-dir358428841/test/extended/testdata/deployments/deployment-simple.yaml'
Mar 29 11:15:26.665: INFO: 00: cancelling deployment
Mar 29 11:15:26.665: INFO: Running 'oc rollout --config=/tmp/extended-test-cli-deployment-xwxwb-4hs6w-user.kubeconfig --namespace=extended-test-cli-deployment-xwxwb-4hs6w cancel dc/deployment-simple'
Mar 29 11:15:27.357: INFO: Error running &{/data/src/github.com/openshift/origin/_output/local/bin/linux/amd64/oc [oc rollout --config=/tmp/extended-test-cli-deployment-xwxwb-4hs6w-user.kubeconfig --namespace=extended-test-cli-deployment-xwxwb-4hs6w cancel dc/deployment-simple] []   error: there have been no replication controllers for extended-test-cli-deployment-xwxwb-4hs6w/deployment-simple
```

**Comment by mfojtik** [comment]
moving to P2 as this haven't occured for couple weeks.

**Comment by mfojtik** [comment]
closing due to age.

**Comment by miminar** [comment]
Happened here: https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_gce/5101/testReport/junit/(root)/Extended/deploymentconfigs_when_run_iteratively__Conformance__should_only_deploy_the_last_deployment/

**Comment by bparees** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_gce/5128/

**Comment by smarterclayton** [comment]
Fixed in #15904

**Comment by bparees** [comment]
@jim-minter possibly relevant to your observations about multiple deployments?
